### PR TITLE
Add-eslint-airbnb-config

### PR DESCRIPTION
**Pull request by ManpreetSL**
#5

#25 removes the Profile page, so that work isn't redone here.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,3 +1,29 @@
...
-  "extends": "next/core-web-vitals"
+    "env": {
+        "browser": true,
+        "es2021": true
+    },
+    "extends": [
+        "airbnb",
+        "airbnb-typescript",
+        "next",
+        "prettier"
```
Comment (Harjot1Singh) [inline .eslintrc.json:10]: Be careful with mixing prettier in like this! I believe it turns off eslint rules that conflict with prettier, but I've always found that running prettier and eslint on the same file can be a pain!

File: .eslintrc.json [hunk 2]
```diff
@@ -1,3 +1,29 @@
...
+    ],
+    "parserOptions": {
+        "ecmaVersion": "latest",
+        "sourceType": "module",
+        "project": "./tsconfig.json"
+    },
+    "plugins": [
+    ],
+    "rules": {
+      "react/react-in-jsx-scope": "off",
```
Comment (Harjot1Singh) [inline .eslintrc.json:22]: Perfect - this is what you need for React 17 onwards until they update the rule (which they'll do once R17+ becomes more mainstream)

File: package.json [hunk 3]
```diff
@@ -12,11 +12,26 @@
...
+    "@typescript-eslint/parser": "^5.0.0",
+    "eslint": "^8.2.0",
+    "eslint-config-airbnb": "^19.0.4",
+    "eslint-config-airbnb-typescript": "^17.0.0",
+    "eslint-config-next": "13.0.6",
+    "eslint-config-prettier": "^8.5.0",
+    "eslint-config-standard-with-typescript": "^23.0.0",
+    "eslint-plugin-import": "^2.25.3",
+    "eslint-plugin-jsx-a11y": "^6.5.1",
+    "eslint-plugin-n": "^15.6.0",
```
Comment (Harjot1Singh) [inline package.json:30]: [curious] what is this for? Never come across it before

File: package.json [hunk 4]
```diff
@@ -12,11 +12,26 @@
...
+  },
+  "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "^5.13.0",
+    "@typescript-eslint/parser": "^5.0.0",
+    "eslint": "^8.2.0",
+    "eslint-config-airbnb": "^19.0.4",
+    "eslint-config-airbnb-typescript": "^17.0.0",
+    "eslint-config-next": "13.0.6",
+    "eslint-config-prettier": "^8.5.0",
+    "eslint-config-standard-with-typescript": "^23.0.0",
```
Comment (Harjot1Singh) [inline package.json:27]: Do double check you're using this - I couldn't see that you are in eslintrc!

File: package.json [hunk 5]
```diff
@@ -12,11 +12,26 @@
...
-    "typescript": "4.9.3"
+    "react-dom": "18.2.0"
+  },
+  "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "^5.13.0",
+    "@typescript-eslint/parser": "^5.0.0",
+    "eslint": "^8.2.0",
+    "eslint-config-airbnb": "^19.0.4",
+    "eslint-config-airbnb-typescript": "^17.0.0",
+    "eslint-config-next": "13.0.6",
```
Comment (Harjot1Singh) [inline package.json:25]: Does this need an explicit install, or does this come with next.js already?

**Review by Harjot1Singh — COMMENTED** [review]
Nice, you've mostly got this - a few comments and curiosities

File: .eslintrc.json (hunk 1 above)
Comment (ManpreetSL) [reply .eslintrc.json:10]: I followed a guide about this and they said it'd be okay to mix the 2, but I'll bear that in mind for the future.

For now, should I remove Prettier?

Comment (Harjot1Singh) [reply .eslintrc.json:10]: Yep, remove it for now!

File: package.json (hunk 5 above)
Comment (ManpreetSL) [reply package.json:25]: I think I installed these myself. Can we go through which ones I actually need?

File: .eslintrc.json (hunk 1 above)
Comment (ManpreetSL) [reply .eslintrc.json]: Done this :)

**Comment by ManpreetSL** [comment]
@Harjot1Singh I've fixed the issues described above and in our call yesterday.

File: package.json (hunk 4 above)
Comment (ManpreetSL) [reply package.json:35]: The extra ones that weren't being used have been removed now.

File: package.json (hunk 3 above)
Comment (ManpreetSL) [reply package.json]: Got these from a tutorial whilst trying to get things working, but `eslint-plugin-import` and `eslint-plugin-n` have been removed now. I think the others are peer dependencies.

File: package.json (hunk 4 above)
Comment (Harjot1Singh) [reply package.json:35]: I think this package isn't used/necessary, from looking at the rules

**Review by Harjot1Singh — APPROVED** [review]
Looks good! I think there's one package you can remove still

File: package.json (hunk 4 above)
Comment (ManpreetSL) [reply package.json:35]: On call, Harjot said `eslint config standard with typescript` is  the one that isn't used. Will remove it.

**Comment by ManpreetSL** [comment]
> remove unused eslint-config-standard-with-typescript package

[remove unused eslint-config-standard-with-typescript package](https://github.com/ManpreetSL/habit-tracker/pull/27/commits/4fb34137062d6fdcf5c9c4547e7f0ef403ba108f)

**Pull request merged**